pl_bolts/datasets/emnist_dataset.py

```diff
@@ -12,6 +12,75 @@
 else:  # pragma: no cover
     warn_missing_pkg('PIL', pypi_name='Pillow')
 
+_TABLE = """
+
+| Name     | Classes | No. Training | No. Testing | Validation | Total   |
+|----------|---------|--------------|-------------|------------|---------|
+| By_Class | 62      | 697,932      | 116,323     | No         | 814,255 |
+| By_Merge | 47      | 697,932      | 116,323     | No         | 814,255 |
+| Balanced | 47      | 112,800      | 18,800      | Yes        | 131,600 |
+| Digits   | 10      | 240,000      | 40,000      | Yes        | 280,000 |
+| Letters  | 37      | 88,800       | 14,800      | Yes        | 103,600 |
+| MNIST    | 10      | 60,000       | 10,000      | Yes        | 70,000  |
+
+"""
+
+EMNIST_METADATA = {
```

Comment:
How will this `EMNIST_METADATA` be used? Is there any use case?

Reply:
> ## Concern(s) about current implementation: :question: 
> 
> - The validation data does not exist for `byclass` and `bymerge` splits, as indicated in the [paper, *table-II*](https://arxiv.org/pdf/1702.05373.pdf). So, should there be a check for these `splits`, where if the user specifies to use no validation data, then the code will not produce any validation dataset and datamodule?
> 
>   - one way this could be done is by introducing another keyword argument: `strict_val_split: bool = False` in the `__init__` method of `EMNISTDataModule` and then check if the metadata for that split has `validation = False`.

Sorry I missed your comment above.

Reply:
Just had a look at the paper. It says:
> Table II contains a summary of the EMNIST datasets and
indicates which classes contain a validation subset in the
training set. In these datasets, the last portion of the training
set, equal in size to the testing set, is set aside as a validation
set. Additionally, this subset is also balanced such that it
contains an equal number of samples for each task. If the
validation set is not to be used, then the training set can be
used as one contiguous set.

As far as I understand, it's just saying that "Validation" in the table means that datasets are balanced or not per class. I think we can just ignore the table and make training and validation sets from torchvision's train set, which is the same as other existing datasets like MNIST. no?

Reply:
@akihironitta Let me take a look at it again and get back to you on this one.

Reply:
I think, after taking another look at the table caption, where `validation = yes`, 

- `len(validation_dataset) == len(test_dataset) == N` 
- `train_dataset[-N:] == validation_dataset`
- balanced datasets: `val` and `test`

On the other hand, where `validation = no`, there is no specification on the size or "balancedness" of the validation dataset. I suggest that the user should be given an option to choose one from the following:

- Create train/test and validation datasets extactly as defined in the paper
- Create train/test and validation datasets to some specification: example -- `val_split = 0.2`

However, if necessary, this change could be introduced as a separate PR later (may be).

What do you think?

Reply:
That sounds reasonable! It'd be great if you could do it in this PR.

Reply:
@akihironitta This `val_split` logic has been implemented now for both `EMNISTDataModule` and `BinaryEMNISTDataModule`, along with the test.

Reply:
@sugatoray Could you remove unused keys, such as "num_total"?